In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.preprocessing import LabelEncoder

In [2]:
dataset=pd.read_csv('PP_Mtrls_and_their_Mchncl_Prprts.csv')



dataset.drop('Material', axis=1, inplace=True)


columns_to_drop = ['Sy','mu']
dataset = dataset.drop(columns=columns_to_drop)



dataset['Use'] = dataset['Use'].astype('object')


dataset=pd.get_dummies(dataset,drop_first=True)

In [3]:
label_encoder = LabelEncoder()

# Fit and transform the target column
dataset['Use_True'] = label_encoder.fit_transform(dataset['Use_True'])
dataset

Su       E      G    Ro  Use_True
0     421  207000  79000  7860         1
1     424  207000  79000  7860         1
2     386  207000  79000  7860         1
3     448  207000  79000  7860         1
4     441  207000  79000  7860         1
..    ...     ...    ...   ...       ...
797   980  206000  80000  7860         0
798  1100  206000  80000  7860         0
799  1200  206000  80000  7860         0
800  1030  206000  80000  7860         0
801  1226  206000  80000  7860         0

[802 rows x 5 columns]

In [4]:
indep=dataset[['Su','E','G','Ro']]
dep=dataset["Use_True"]

#split into training set and test
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(indep, dep, test_size = 1/3, random_state = 0)

In [5]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [6]:
su_input=float(input("Ultimate Tensile Strength (Su) in MPa:"))
E_input=float(input("Elastic Modulus (E) in MPa:"))
g_input=float(input("Shear Modulus (G) in MPa:"))
ro_input=float(input("Density (Ro) in Kg/m3:"))

Ultimate Tensile Strength (Su) in MPa:1000
Elastic Modulus (E) in MPa:119876
Shear Modulus (G) in MPa:67543
Density (Ro) in Kg/m3:8912


In [7]:
ppinput=sc.transform([[su_input,E_input,g_input,ro_input]])
ppinput

/Users/viswanathanmuthu/anaconda3/envs/AIVE_Course/lib/python3.11/site-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names
  warnings.warn(


array([[ 1.01737213, -0.6495612 ,  0.01127766,  1.12379576]])

In [8]:
import pickle
loaded_model=pickle.load(open("finalized_model_Random_Forest.sav",'rb'))
y=loaded_model.predict(ppinput)
if (y==0):
    print("Future_Prediction=[False]")
else:
    print('Future_Prediction=[True]')

Future_Prediction=[False]


Ultimate Tensile Strength (Su) in MPa:1000
Elastic Modulus (E) in MPa:119876
Shear Modulus (G) in MPa:67543
Density (Ro) in Kg/m3:8912